pysat/__init__.py

```diff
@@ -79,20 +79,29 @@
     # user modules file
     with open(os.path.join(pysat_dir, 'user_modules.txt'), 'w') as f:
         f.write('')
-        user_modules = []
+        user_modules = {}
 
 else:
     # load up stored data path
     with open(os.path.join(pysat_dir, 'data_path.txt'), 'r') as f:
         data_dir = f.readline()
+
     # load up stored user modules
-    user_modules = []
+    user_modules = {}
     modules_file = os.path.join(pysat_dir, 'user_modules.txt')
     if os.path.exists(modules_file):
-        with open(modules_file, 'r') as f:
-            for _ in f:
-                if _ != '' and (_ is not None):
-                    user_modules.append(_.strip())
+        with open(modules_file, 'r') as fopen:
+            for line in fopen:
+                if line != '' and (line is not None):
+                    # remove trailing whitespace
+                    line = line.strip()
+                    # stored as platform, name, module string
+                    platform, name, inst_module = line.split(' ')
+                    # dict of dicts, keyed by platform then name
+                    if platform not in user_modules:
+                        user_modules[platform] = {}
+                    # store instrument module string
+                    user_modules[platform][name] = inst_module
     else:
         # write user modules file
         with open(os.path.join(pysat_dir, 'user_modules.txt'), 'w') as f:
```

Comment:
Fix single character variable name here and elsewhere.

Reply:
Those changes were in #376, and now #525, and in progress.